File: top-interview-150/0045-jump-game-ii/solution.py
# 45. Jump Game II
# https://leetcode.com/problems/jump-game-ii/
# Accepted: 2026-09-03T10:38:38.000Z
# Language: Python3
# Collection: top-interview-150
# Runtime: 0 ms · Beats 100%
# Memory: 20.1 MB · Beats 65.89%
# Submission: https://leetcode.com/submissions/detail/2129507260/

class Solution:
    def jump(self, nums: List[int]) -> int:
        current_end = 0
        max_portee = nums[0]
        jumps = 0
        n = len(nums)
        for i in range(n-1) : 
            max_portee = max(max_portee, i+nums[i])
            if i == current_end : 
                jumps +=1
                current_end = max_portee
        
        return jumps
            

File: top-interview-150/0045-jump-game-ii/solution.cpp
// 45. Jump Game II
// https://leetcode.com/problems/jump-game-ii/
// Accepted: 2026-09-25T06:28:44.000Z
// Language: C++
// Collection: top-interview-150
// Runtime: 0 ms · Beats 100%
// Memory: 20.6 MB · Beats 78%
// Submission: https://leetcode.com/submissions/detail/2152705700/

class Solution {
public:
    int jump(vector<int>& nums) {
        int portee = nums[0] ; 
        int current_end = 0 ; 
        int jumps = 0 ; 
        for (int i = 0 ;  i< nums.size() -1; i++){
            portee = max(portee, i + nums[i]) ; 
            if (current_end == i){
                jumps +=1 ; 
                current_end = portee ; 
            }
        }
        return jumps ;
    }
};
